internal/systemd/systemd.go

```diff
@@ -0,0 +1,174 @@
+		if err != nil {
+			return nil, fmt.Errorf("cannot establish system connection to systemd: %v", err)
+		}
+	} else {
+		conn, err = systemd.NewUserConnectionContext(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("cannot establish user connection to systemd: %v", err)
+		}
+	}
+
+	return &Conn{
+		ctx:  ctx,
+		conn: conn,
+	}, nil
+}
+
+func (c *Conn) Close() {
+	c.conn.Close()
+}
+
+// Reload instructs systemd to scan for and reload unit files.
+func (c *Conn) Reload() error {
+	return c.conn.ReloadContext(c.ctx)
+}
+
+// EnableUnit enables the named unit. If activate is true, it also starts the
+// unit. If runtime is true, the unit is enabled for the runtime only (/run). If
+// false, it is enabled persistently (/etc).
+func (c *Conn) EnableUnit(name string, activate bool, runtime bool) error {
+	if _, _, err := c.conn.EnableUnitFilesContext(c.ctx, []string{name}, runtime, true); err != nil {
+		return fmt.Errorf("cannot enable unit %v: %v", name, err)
+	}
+
+	if activate {
+		if err := c.StartUnit(name, true); err != nil {
+			return fmt.Errorf("cannot start unit %v: %v", name, err)
+		}
+	}
+
+	return nil
+}
+
+// StartUnit starts the named unit. If wait is true, the method waits until the
+// unit state becomes "active".
+func (c *Conn) StartUnit(name string, wait bool) error {
+	jobComplete := make(chan string)
+	_, err := c.conn.StartUnitContext(c.ctx, name, "replace", jobComplete)
+	if err != nil {
+		return fmt.Errorf("cannot start unit %v: %v", name, err)
+	}
+	result := <-jobComplete
+	switch result {
+	case "done":
+		// The job successfully started, break to proceed
+		break
+	default:
+		return fmt.Errorf("failed to start unit with reason: %v", result)
+	}
+
+	if wait {
+		if err := c.waitForState(name, "active", 1*time.Second); err != nil {
+			return fmt.Errorf("timed out waiting for state 'active': %v", err)
+		}
+	}
+
+	return nil
+}
+
+// DisableUnit disables the named unit. If deactivate is true, it also stops the
+// unit. If runtime is true, the unit is enabled for the runtime only (/run). If
+// false, it is enabled persistently (/etc).
+func (c *Conn) DisableUnit(name string, deactivate bool, runtime bool) error {
+	if _, err := c.conn.DisableUnitFilesContext(c.ctx, []string{name}, runtime); err != nil {
+		return fmt.Errorf("cannot disable unit %v: %v", name, err)
+	}
+
+	if deactivate {
+		if err := c.StopUnit(name, true); err != nil {
+			return fmt.Errorf("cannot stop unit %v: %v", name, err)
+		}
+	}
+
+	return nil
+}
+
+// StopUnit stops the named unit. If wait is true, the method waits until the
+// unit state becomes "inactive".
+func (c *Conn) StopUnit(name string, wait bool) error {
+	jobComplete := make(chan string)
+	_, err := c.conn.StopUnitContext(c.ctx, name, "replace", jobComplete)
+	if err != nil {
+		return fmt.Errorf("cannot stop unit %v: %v", name, err)
+	}
+	result := <-jobComplete
+	switch result {
+	case "done":
+		// The job successfully started, break to proceed
```

Comment:
This comment is stating the obvious isn't it? Coming from Ruby in my past, where comments are frowned upon, I have come to the opinion that comments are valuable when the code is not self documenting. In this case I personally feel the code is self documenting and the comment can be omitted. 

Just a suggestion. Up to you...

Reply:
This is very true. I guess I didn't like the fact that we have an explicit case
without any text beside a `break`, but you're absolutely right and I agree with
you that comments are only useful when the code does not clearly comment itself.